Use `#ifdef` instead of `#if`

We're getting the following error:

```
include/pressio-log/./logger/logger.hpp:59:27: error: expected value in expression
#if PRESSIO_ENABLE_TPL_MPI
```

To fix this, let's just check if a macro is defined (`#ifdef`), without checking the value (`#if`). Documentation will have to be updated to reflect this change, and we should re-release 0.15.0 once this is complete.

--- a/README.md
+++ b/README.md
@@ -67,15 +67,15 @@ git clone https://github.com/Pressio/pressio-log.git
 You can define any of the following macros before including the core pressio-log file:
 
 ```cpp
-#define PRESSIO_SILENCE_WARNINGS 1
-#define PRESSIO_ENABLE_COLORIZED_OUTPUT 1
-#define PRESSIO_ENABLE_EXTERNAL_FMT 1
-#define PRESSIO_ENABLE_TPL_MPI 1
+#define PRESSIO_SILENCE_WARNINGS
+#define PRESSIO_ENABLE_COLORIZED_OUTPUT
+#define PRESSIO_ENABLE_EXTERNAL_FMT
+#define PRESSIO_ENABLE_TPL_MPI
 ```
 
 If you enable an external fmt, make sure that your project properly includes the `fmt` library you would like to use.
 
-By default, the above options are all turned off, and pressio-log uses the in-house `fmt` snapshotted in `include/fmt`.
+By default, the above options are not defined, and pressio-log uses the in-house `fmt` snapshotted in `include/fmt`.
 
 3. **Include** the library
 
@@ -152,7 +152,7 @@ PRESSIOLOG_SET_COMMUNICATOR(MPI_Comm comm);
 ### Sample Program
 
 ```cpp
-#define PRESSIOLOG_COLORIZE_OUTPUT 1
+#define PRESSIOLOG_COLORIZE_OUTPUT
 
 #include <pressio-log/core.hpp>
 
@@ -175,7 +175,7 @@ int main() {
 
 ### Using With pressio-rom
 
-If the `PRESSIO_ENABLE_LOGGING` macro is set to `1`, `pressio-rom` will automatically look for the
+If the `PRESSIO_ENABLE_LOGGING` macro is defined`, `pressio-rom` will automatically look for the
 `pressio-log` library and include the `pressio-log/core.hpp` file.
 
 This means that as long as you have pointed your app to the `pressio-log` include directory, you **do not** need to explicitly include any pressio-log files.
@@ -189,7 +189,7 @@ The following example initializes the logger with LogLevel `info` and directs th
 In practice, this means that all solver output (level: `sparse`) and critical program information (level: `info`) from Pressio will be written to that file. Any warnings or errors will also be logged, while all other output (level: `debug`) will be omitted.
 
 ```cpp
-#define PRESSIO_ENABLE_LOGGING 1
+#define PRESSIO_ENABLE_LOGGING
 
 #include <pressio/...>
 

--- a/include/pressio-log/core.hpp
+++ b/include/pressio-log/core.hpp
@@ -66,7 +66,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 // Initialization/Finalization
 
-#if PRESSIO_ENABLE_TPL_MPI
+#ifdef PRESSIO_ENABLE_TPL_MPI
     #define PRESSIOLOG_INITIALIZE(...) \
         pressiolog::Logger::PressioLogger()->initializeWithMPI(__VA_ARGS__)
 #else
@@ -89,7 +89,7 @@
 #define PRESSIOLOG_SET_OUTPUT_FILENAME(...) \
     pressiolog::Logger::PressioLogger()->setOutputFilename(__VA_ARGS__)
 
-#if PRESSIO_ENABLE_TPL_MPI
+#ifdef PRESSIO_ENABLE_TPL_MPI
 
     #define PRESSIOLOG_SET_LOGGING_RANK(...) \
         pressiolog::Logger::PressioLogger()->setLoggingRank(__VA_ARGS__)

--- a/include/pressio-log/fmt/fmt.h
+++ b/include/pressio-log/fmt/fmt.h
@@ -6,7 +6,7 @@
 
 #define FMT_HEADER_ONLY
 
-#if PRESSIO_ENABLE_EXTERNAL_FMT
+#ifdef PRESSIO_ENABLE_EXTERNAL_FMT
     #include <fmt/format.h>
 #else
     #include <pressio-log/fmt/format.h>

--- a/include/pressio-log/logger/colors.hpp
+++ b/include/pressio-log/logger/colors.hpp
@@ -54,7 +54,7 @@
 namespace pressiolog { namespace colors {
 
 inline std::string color(std::string str, std::string color_code) {
-    #if PRESSIO_ENABLE_COLORIZED_OUTPUT
+    #ifdef PRESSIO_ENABLE_COLORIZED_OUTPUT
     return "\033[" + color_code + "m" + str + "\033[0m";
     #endif
     return str;

--- a/include/pressio-log/logger/logger.hpp
+++ b/include/pressio-log/logger/logger.hpp
@@ -56,7 +56,7 @@
 #include <filesystem>
 
 #include <pressio-log/fmt/fmt.h>
-#if PRESSIO_ENABLE_TPL_MPI
+#ifdef PRESSIO_ENABLE_TPL_MPI
 #include <mpi.h>
 #endif
 
@@ -87,7 +87,7 @@ class Logger {
             LogTo destination = LogTo::console,
             const std::string& filename = "pressio.log"
         );
-        #if PRESSIO_ENABLE_TPL_MPI
+        #ifdef PRESSIO_ENABLE_TPL_MPI
         void initializeWithMPI(
             LogLevel level = LogLevel::sparse,
             LogTo destination = LogTo::console,
@@ -107,7 +107,7 @@ class Logger {
         void setLoggingLevel(LogLevel level);
         void setOutputStream(LogTo destination);
         void setOutputFilename(const std::string& log_file_name);
-        #if PRESSIO_ENABLE_TPL_MPI
+        #ifdef PRESSIO_ENABLE_TPL_MPI
         void setLoggingRank(int rank);
         void setCommunicator(MPI_Comm comm);
         #endif
@@ -126,7 +126,7 @@ class Logger {
         void assertLoggerIsInitialized_();
 
         // MPI helpers
-        #if PRESSIO_ENABLE_TPL_MPI
+        #ifdef PRESSIO_ENABLE_TPL_MPI
         void updateCurrentRank_();
         #endif
 
@@ -175,7 +175,7 @@ class Logger {
         std::string log_file_{"pressio.log"};
 
         // MPI
-        #if PRESSIO_ENABLE_TPL_MPI
+        #ifdef PRESSIO_ENABLE_TPL_MPI
         bool mpi_initialized_{false};
         MPI_Comm comm_{MPI_COMM_WORLD};
         #endif

--- a/include/pressio-log/logger/logger_impl.hpp
+++ b/include/pressio-log/logger/logger_impl.hpp
@@ -75,7 +75,7 @@ inline void Logger::initialize(
     }
 }
 
-#if PRESSIO_ENABLE_TPL_MPI
+#ifdef PRESSIO_ENABLE_TPL_MPI
 inline void Logger::initializeWithMPI(
     LogLevel level, LogTo destination, const std::string& filename,
     int logging_rank, MPI_Comm comm) {
@@ -151,7 +151,7 @@ inline void Logger::setOutputFilename(const std::string& log_file_name) {
     log_file_ = log_file_name;
 }
 
-#if PRESSIO_ENABLE_TPL_MPI
+#ifdef PRESSIO_ENABLE_TPL_MPI
 inline void Logger::setLoggingRank(int rank) {
     if (mpi_initialized_) {
         int size;
@@ -178,7 +178,7 @@ inline Logger::Logger() {}
 ///////////////////////////////////////////////////////////////////////////////
 // MPI helpers
 
-#if PRESSIO_ENABLE_TPL_MPI
+#ifdef PRESSIO_ENABLE_TPL_MPI
 inline void Logger::updateCurrentRank_() {
     if (mpi_initialized_) {
         MPI_Comm_rank(comm_, &current_rank_);
@@ -228,7 +228,7 @@ inline void Logger::error_(const std::string& message) {
     }
 }
 inline void Logger::warning_(const std::string& message) {
-    #if not PRESSIO_SILENCE_WARNINGS
+    #ifndef PRESSIO_SILENCE_WARNINGS
     if (logging_level_ >= LogLevel::warning) {
         log_(formatWarning_(message));
     }

--- a/tests/cmake/options.cmake
+++ b/tests/cmake/options.cmake
@@ -3,16 +3,12 @@
 option(PRESSIO_SILENCE_WARNINGS "Enable or disable warnings" OFF)
 if (PRESSIO_SILENCE_WARNINGS)
   add_compile_definitions(PRESSIO_SILENCE_WARNINGS=1)
-else ()
-  add_compile_definitions(PRESSIO_SILENCE_WARNINGS=0)
 endif ()
 message(">> PRESSIO_SILENCE_WARNINGS = ${PRESSIO_SILENCE_WARNINGS}")
 
 option(PRESSIO_ENABLE_COLORIZED_OUTPUT "Enable or disable colorized logging" OFF)
 if (PRESSIO_ENABLE_COLORIZED_OUTPUT)
   add_compile_definitions(PRESSIO_ENABLE_COLORIZED_OUTPUT=1)
-else ()
-  add_compile_definitions(PRESSIO_ENABLE_COLORIZED_OUTPUT=0)
 endif ()
 message(">> PRESSIO_ENABLE_COLORIZED_OUTPUT = ${PRESSIO_ENABLE_COLORIZED_OUTPUT}")
 
@@ -44,8 +40,6 @@ if (PRESSIO_ENABLE_EXTERNAL_FMT)
     message(WARNING "PRESSIO_ENABLE_EXTERNAL_FMT is ON, but fmt was not found. "
                     "Use fmt_INCLUDE_DIR, fmt_ROOT, or fmt_DIR to provide the correct path.")
   endif ()
-else ()
-  add_compile_definitions(PRESSIO_ENABLE_EXTERNAL_FMT=0)
 endif ()
 
 # mpi
@@ -56,6 +50,4 @@ if (PRESSIO_ENABLE_TPL_MPI)
   find_package(MPI REQUIRED)
   link_libraries(${MPI_CXX_LIBRARIES})
   include_directories(SYSTEM ${MPI_CXX_INCLUDE_DIRS})
-else ()
-  add_compile_definitions(PRESSIO_ENABLE_TPL_MPI=0)
 endif ()

